lib/cook_book.rb

```diff
@@ -0,0 +1,31 @@
+class CookBook
+  attr_reader :recipes
+
+  def initialize
+    @recipes = []
+  end
+
+  def add_recipe(recipe)
+    @recipes << recipe
+  end
+
+  def ingredients
+    list = []
+    @recipes.each do |recipe|
+      recipe.ingredients.each do |ingredient|
+        list << ingredient.name
+      end
+    end
+    list.uniq
```

Comment:
Great job! This works. I'd love to see if you could do this with a different enumerable too.